sphinx_autodoc_typehints.py

```diff
@@ -1,40 +1,126 @@
-import inspect
+import sys
 
 from sphinx.util.inspect import getargspec
 from sphinx.ext.autodoc import formatargspec
 
 try:
-    from backports.typing import Optional, get_type_hints
+    from backports.typing import (Any, Callable, Generic, GenericMeta, Tuple, TypeVar, TypingMeta,
+                                  Union, _ForwardRef, _TypeAlias, get_type_hints)
 except ImportError:
-    from typing import Optional, get_type_hints
+    from typing import (Any, Callable, Generic, GenericMeta, Tuple, TypeVar, TypingMeta,
+                        Union, _ForwardRef, _TypeAlias, get_type_hints)
 
+try:
+    from typing import ClassVar
+except ImportError:
+    ClassVar = None
+
+
+def format_annotation(annotation, obj=None):
+    if isinstance(annotation, type):
+        qname = sys.version_info[:2] >= (3, 3) and annotation.__qualname__ or annotation.__name__
```

Comment:
Why this dance with `__qualname__`? The project only supports Python 3.3+ anyway, as is evident from the classifiers in setup.py.